lib/SILOptimizer/Transforms/OptRemarkGenerator.cpp

```diff
@@ -41,6 +41,15 @@ static llvm::cl::opt<bool> ForceVisitImplicitAutogeneratedFunctions(
         "Emit opt remarks even on implicit and autogenerated functions"),
     llvm::cl::init(false));
 
+static llvm::cl::opt<bool> DecllessDebugValueUseSILDebugInfo(
+    "optremarkgen-declless-debugvalue-use-sildebugvar-info", llvm::cl::Hidden,
+    llvm::cl::desc(
+        "If a debug_value does not have a decl, infer a value with a name from "
+        "that info that has a loc set to the loc of the debug_value "
+        "instruction itself. This is for testing purposes so it is easier to "
+        "write SIL test cases for this pass"),
```

Comment:
This doesn't look quite right to me. You are not supposed to extaract the variable name from the Decl. All instruction that carry variable debug info (`debug_value`, `debug_value_addr`, `alloc_stack`, `alloc_box`, ...?) have a `getVarInfo()` method that should be used instead.

Reply:
see https://github.com/apple/swift/blob/bf7c03b1f6cce062e27684abe5018a723a594662/include/swift/SIL/SILInstruction.h#L1467 for example.

Reply:
Ok. Let me fix that.

Reply:
I think the option is still needed since the loc that I need is from the decl.